### ISSUE TITLE
PerferVideoOnly in VideoTextureProvider

### ISSUE THREAD

**Issue by kazu0617** [Yellow-Dog-Man/Resonite-Issues#3206]
### Is your feature request related to a problem? Please describe.

Curently, due to Youtube playback system, it's only returns 360p video only if combined.
but 360p is so badly graphics in VR environment.

### Describe the solution you'd like

I want to check "PerferVideoOnly" in VideoTextureProvider. it's much more good graphics quality. 

※but without songs, so i'll also use PreferAudioOnly if want to listen music.

### Describe alternatives you've considered

if resonite combine audio and video from VideoTextureProvider is much more great, but so hevily development cost.

### Additional Context

you can check all youtube videos, so plz check youtube videos. for example: https://www.youtube.com/watch?v=nrf74Ow8oEM

### Requesters

all resonite name.
kazu, ChameleonO2, alistair442

**Comment by shiftyscales** [comment]
IIRC we currently / historically cap video quality at 720p due to hardware acceleration being disabled in the existing playback engines (as enabling hardware acceleration causes crashing, etc.) so there wouldn't be a ton of immediate benefit to fetching higher quality video only streams- as the stream we'd fetch is still comparatively low to what's available.

So it would probably be better if we fix the underlying issue if possible to display a higher quality stream (or implement some method to remux higher quality audio/video streams) to start since most of your concern seems to be based around the recent regression in video quality from yt-dlp.

> but so hevily development cost.

It's best to focus on reporting what you **need** and leave assessments of implementation difficulty to our coders, @kazu0617. Based on the alternative you provided, it sounds as though you'd prefer if remuxing higher quality streams were an option.

Curious for input from @Frooxius otherwise- passing the `bv` and `ba` arguments to yt-dlp would allow us to download the best quality video / audio streams, but if the user has ffmpeg installed, it's also possible for it to remux them.

> You can merge the video and audio of multiple formats into a single file using -f <format1>+<format2>+... (requires ffmpeg installed); e.g. -f bestvideo+bestaudio will download the best video-only format, the best audio-only format and mux them together with ffmpeg.

https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#format-selection

**Comment by kazu0617** [comment]
> It's best to focus on reporting what you need and leave assessments of implementation difficulty to our coders, @kazu0617. Based on the alternative you provided, it sounds as though you'd prefer if remuxing higher quality streams were an option.

Yes, I agree. I think it would be best if it were possible to remix.

However, to remix, it is not enough to change only the “coding,” but it will be necessary to modify the design in the background, which is why I mentioned “development costs seem to be heavy.
There are two approaches I can think of right off the top of my head.

- Use yt-dlp to get the bestvideo / bestaudio parameters respectively. Stream each of them and match the timestamps in Resonite.
- When using yt-dlp, provide a full download option in addition to the streaming method, and then play it back.

Each has its advantages and disadvantages, and each requires more development time.


Considering the above, I thought “PreferVideoOnly” could be implemented with the least man-hours possible (since there is already “PreferAudioOnly”), so I decided to start with PreferVideoOnly... and raised this Issue.
